addons/mail/models/mail_thread.py

```diff
@@ -4546,3 +4546,17 @@ def _generate_tracking_message(self, message, return_line='\n'):
             tracking_message += return_line
 
         return tracking_message
+
+    def _merge_method(self, destination, source):
+        """ Using the default merge method risks duplicating the SQL key for mail_followers (res_model, res_id, partner_id).
+        Adding a partner as follower to a record twice nullifies the transaction, preventing the merge of followers.
+        This function ensures we don't try unsafe subscriptions.
+        """
+        if not hasattr(destination, 'message_follower_ids') or not hasattr(source, 'message_follower_ids'):
+            return
```

Comment:
can we different models that are not mail.thread (or inheriting mail.thread)? if not this doesn't seem necessary since we are in mail.thread model code that should always have "message_follower_ids" field

Reply:
You're right, that check should really be in data_merge instead I'm not sure what I was thinking

Reply:
Or not, data merge apparently depends on mail. I'll fix the commit right away :+1:

Reply:
@nle-odoo Done